### PR TITLE
[test/integration/pod] Setup all indexes.

### LINKED ISSUE #1258: Some integration tests don't set all indexes
**What happened**:

In integration tests logs, you might find the following errors:
```
Index with name field:.metadata.parentWorkload does not existUnable to list child jobs
```

```
  2023-10-25T19:42:43.924182458Z	ERROR	workload-reconciler	workload/resources.go:112	Failed adjusting requests for LimitRanges	{"workload": {"name":"pod-test-pod-9fd67","namespace":"pod-namespace-wxtrr"}, "queue": "test-queue", "status": "admitted", "clusterQueue": "cluster-queue", "error": "Index with name field:spec.hasContainerType does not exist"}
```

Although the test succeeds (probably because the test doesn't depend on these reconcilers), this pollutes the logs.

**What you expected to happen**:

No errors related to indexes

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Setup the core indexes in the pod integration tests.

#### Which issue(s) this PR fixes:

Fixes #1258

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/jobs/pod/suite_test.go [hunk 1]
```diff
@@ -63,7 +63,10 @@ func TestAPIs(t *testing.T) {
 
 func managerSetup(opts ...jobframework.Option) framework.ManagerSetup {
 	return func(mgr manager.Manager, ctx context.Context) {
-		err := pod.SetupIndexes(ctx, mgr.GetFieldIndexer())
+		err := indexer.Setup(ctx, mgr.GetFieldIndexer())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		err = pod.SetupIndexes(ctx, mgr.GetFieldIndexer())
```
Comment (tenzen-y) [inline test/integration/controller/jobs/pod/suite_test.go:69]: Doesn't this flaky happen in other integrations?

In other integrations, we don't set up indexer such as batch/job:

https://github.com/kubernetes-sigs/kueue/blob/0602e146fa8fd01bf83cb368acb027041ec1f011/test/integration/controller/jobs/job/suite_test.go#L62-L75

**Review by tenzen-y — COMMENTED** [review]
Generally, lgtm.

File: test/integration/controller/jobs/pod/suite_test.go (hunk 1 above)
Comment (trasc) [reply test/integration/controller/jobs/pod/suite_test.go:69]: Other suites don't use the core controllers and batch job controller.

Comment (tenzen-y) [reply test/integration/controller/jobs/pod/suite_test.go:69]: It makes sense. Thanks.

**Comment by tenzen-y** [comment]
cc: @alculquicondor

**Pull request merged**